packages/web-components/src/components/ic-side-navigation/ic-side-navigation.tsx

```diff
@@ -345,23 +345,34 @@ export class SideNavigation {
       ".bottom-wrapper > .secondary-navigation"
     );
 
+    const classToRemove =
+      type === "start"
+        ? this.COLLAPSED_ICON_LABELS_END
+        : this.COLLAPSED_ICON_LABELS_START;
+
+    const classToAdd =
+      type === "start"
+        ? this.COLLAPSED_ICON_LABELS_START
+        : this.COLLAPSED_ICON_LABELS_END;
+
     if (primaryNavigationWrapper) {
-      primaryNavigationWrapper.classList.remove(this.COLLAPSED_ICON_LABELS_END);
-      primaryNavigationWrapper.classList.add(this.COLLAPSED_ICON_LABELS_START);
+      primaryNavigationWrapper.classList.remove(classToRemove);
+      primaryNavigationWrapper.classList.add(classToAdd);
     }
 
     if (secondaryNavigationWrapper) {
-      secondaryNavigationWrapper.classList.remove(
-        this.COLLAPSED_ICON_LABELS_END
-      );
-      secondaryNavigationWrapper.classList.add(
-        this.COLLAPSED_ICON_LABELS_START
-      );
+      secondaryNavigationWrapper.classList.remove(classToRemove);
+      secondaryNavigationWrapper.classList.add(classToAdd);
     }
   };
 
+  private transitionEndHandler = () => {
+    this.transitionHandler("end");
+  };
+
   private animateCollapsedIconLabels = () => {
-    this.transitionEndHandler();
+    this.transitionHandler("start");
+    this.transitionHandler("end");
```

Comment:
Can use your new `transitionEndHandler` method here..or if it's only being used once, remove it.